lib/java_buildpack/framework/dynatrace_one_agent.rb

```diff
@@ -31,34 +31,41 @@ class DynatraceOneAgent < JavaBuildpack::Component::VersionedDependencyComponent
       def initialize(context)
         super(context)
         @version, @uri = agent_download_url if supports?
+        @logger        = JavaBuildpack::Logging::LoggerFactory.instance.get_logger DynatraceOneAgent
       end
 
       # (see JavaBuildpack::Component::BaseComponent#compile)
       def compile
-        JavaBuildpack::Util::Cache::InternetAvailability.instance.available(
-          true, 'The Dynatrace One Agent download location is always accessible'
-        ) do
-          download(@version, @uri) { |file| expand file }
+        begin
```

Comment:
Ruby has a syntax that allows you to embed the `begin/rescue` into the method declaration.  I'll update in a following commit.